### PR TITLE
fix: createOriginHeaders in case where no origin request header is present

### LINKED ISSUE #1: Incorrect Access-Control-Allow-Origin header when `cors` option is function and no origin header is present
Hello and thank you for making this helpful open source contribution 🙌

I'm testing this out in a Nuxt server-side h3 app and running into an error from h3:

`undefined is not allowed origin`

My situation is that I have defined the origin option as a function that returns a boolean. This works, however, I'm making a request from the browser to the same origin. This means that the browser does not add the `origin` header. So when your package calls `createOriginHeaders()`, it ends up trying to add `Access-Control-Allow-Origin` header as `undefined`.

```ts
export function createOriginHeaders (event: CompatibilityEvent, options: CorsOptions): AccessControlAllowOriginHeader {
  const { origin } = options // origin here is a function 🙈

  if (!origin || origin === '*') {
    return { 'Access-Control-Allow-Origin': '*' }
  }

  if (typeof origin === 'string') {
    return { 'Access-Control-Allow-Origin': origin, Vary: 'Origin' }
  }

  const requestOriginHeader = getRequestHeader(event, 'Origin') as string // requestOriginHeader is actually undefined! 😬
  const headers = Array.isArray(requestOriginHeader) ? requestOriginHeader.join(',') : requestOriginHeader

  return { 'Access-Control-Allow-Origin': headers, Vary: 'Origin' }
}
```

As you are checking the origin function to see if it is allowed before the code gets to this point, I think you can just assume that if `requestOriginHeader` is undefined, then it should be '*'. Or potentially if `requestOriginHeader` is undefined then you should return an empty object?

PR incoming...

### PR DESCRIPTION
**Pull request by JamieCurnow**
Should fix #1

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -58,8 +58,8 @@ export function createOriginHeaders (event: CompatibilityEvent, options: CorsOpt
     return { 'Access-Control-Allow-Origin': origin, Vary: 'Origin' }
   }
 
-  const requestOriginHeader = getRequestHeader(event, 'Origin') as string
```
Comment (nozomuikuta) [inline src/utils.ts:61]: This is a bug of `getRequestHeader` function.
It returns `string | string[]` while it should return `string | string[] | undefined`.

I will create a bug report in the h3 upstream repository and add a reference to it as a comment.

---

Confession: I added the utility functions to the upstream, so it's my mistake.

**Review by nozomuikuta — CHANGES_REQUESTED** [review]
@JamieCurnow 

Hi, thank you for reporting the bug and submitting a PR!!

While I'm reporting the upstream bug, would you add a test case for your PR?

Here is the related part: 
https://github.com/NozomuIkuta/h3-cors/blob/cc04f8579a59c053d163ca695a75ac9849bc95fc/test/utils.test.ts#L167

I appreciate if you add your test case at the end of the "describe" code block.

File: src/utils.ts [hunk 2]
```diff
@@ -58,8 +58,8 @@ export function createOriginHeaders (event: CompatibilityEvent, options: CorsOpt
...
-  const headers = Array.isArray(requestOriginHeader) ? requestOriginHeader.join(',') : requestOriginHeader
+  const requestOriginHeader = getRequestHeader(event, 'Origin') as string | undefined
+  const headers = !requestOriginHeader ? '*' : Array.isArray(requestOriginHeader) ? requestOriginHeader.join(',') : requestOriginHeader
```
Comment (nozomuikuta) [inline src/utils.ts]:
```suggestion
  // Type assertion is needed due to upstream issue: https://github.com/unjs/h3/issues/176
  const requestOriginHeader = getRequestHeader(event, 'Origin') as string | undefined
  const headers = !requestOriginHeader ? '*' : Array.isArray(requestOriginHeader) ? requestOriginHeader.join(',') : requestOriginHeader
```

**Review by nozomuikuta — CHANGES_REQUESTED** [review]

File: src/utils.ts (hunk 1 above)
Comment (nozomuikuta) [reply src/utils.ts:61]: This conversation can be resolved once https://github.com/NozomuIkuta/h3-cors/pull/2#discussion_r966198831 is resolved.

File: src/utils.ts (hunk 2 above)
Comment (nozomuikuta) [reply src/utils.ts]: Applied: 7c1e308dcd7e680a2ede265915006763e3809251

**Review by nozomuikuta — APPROVED** [review]
I added the following changes and approve this PR.

- Fixed PR title to make commit message semantic
- Refactored the logic
- Added a unit test to cover the change

---

@JamieCurnow 

Thank you for contribution!
I will let you know when I publish the package to the npm registry!

**Comment by JamieCurnow** [comment]
@NozomuIkuta thank you so much 👌 I was going to get around to it this weekend but you beat me to it 🙌

**Comment by nozomuikuta** [comment]
@JamieCurnow 

I have just published the package. 🙋‍♀️ 
Please check v0.1.4. 👍

**Pull request merged**